utils/src/types/stages.types.ts

```diff
@@ -27,14 +26,17 @@ export const ALLOWED_STAGE_PROGRESSION = {
   [StageKind.SetProfile]: false,
   [StageKind.GroupChat]: false,
   [StageKind.VoteForLeader]: true,
-  [StageKind.RevealVoted]: false,
+  [StageKind.Reveal]: false,
   [StageKind.TakeSurvey]: true,
-  // [StageKind.RankItems]: false,
 } as const;
 
+
 export interface BaseStageConfig {
+  id: string;
   kind: StageKind;
   name: string;
+  composite?: boolean; // true if stage uses data from other stages (e.g., reveal)
```

Comment:
Nit: `dependsOnOtherStages` or too long?

Reply:
Hmm I don't have a strong preference, though I figured "composite" might be a faster term to refer to it (e.g., "composite stage") throughout the codebase (e.g., define it in documentation and then use everywhere)?